ivy_tests/test_ivy/test_functional/test_core/test_manipulation.py

```diff
@@ -417,37 +425,36 @@ def test_squeeze(
 
 @st.composite
 def _stack_helper(draw):
-    shape = tuple(
-        draw(st.lists(helpers.ints(min_value=1, max_value=10), min_size=0, max_size=5))
-    )
-    axis = draw(helpers.ints(min_value=-len(shape), max_value=len(shape)))
-    num_arrays = draw(
-        st.shared(helpers.ints(min_value=1, max_value=3), key="num_arrays")
-    )
+    shape = draw(st.shared(helpers.get_shape(min_num_dims=1), key="values_shape"))
+    num_arrays = draw(st.shared(st.integers(1, 3), key="num_arrays"))
     dtype = draw(st.sampled_from(ivy_np.valid_dtypes))
-    dtypes_arrays = draw(
-        st.lists(
-            helpers.dtype_and_values(available_dtypes=[dtype], shape=shape),
-            min_size=num_arrays,
-            max_size=num_arrays,
-        )
-    )
-    return dtypes_arrays, axis
+    arrays = []
+    dtypes = [dtype for _ in range(num_arrays)]
+
+    for _ in range(num_arrays):
+        _, array = draw(helpers.dtype_and_values(available_dtypes=[dtype], shape=shape))
+        arrays.append(np.asarray(array, dtype=dtype))
+    return dtypes, arrays
 
 
 # stack
+@settings(deadline=500)
 @given(
-    dtypes_arrays_axis=_stack_helper(),
+    dtypes_arrays=_stack_helper(),
+    axis=helpers.get_axis(
+        shape=st.shared(helpers.get_shape(min_num_dims=1), key="values_shape"),
+        force_int=True,
+    ),
     as_variable=helpers.array_bools(
-        num_arrays=st.shared(helpers.ints(min_value=1, max_value=3), key="num_arrays")
+        num_arrays=st.shared(st.integers(1, 3), key="num_arrays")
```

Comment:
we should be using `helpers.ints` and `helpers.floats`, please make sure this is the case for all tests.

Reply:
Made the changes and pushed them to the branch